pkg/daemon/ceph/osd/encryption.go

```diff
@@ -74,3 +82,59 @@ func setKEKinEnv(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo)
 
 	return nil
 }
+
+func setLUKSLabelAndSubsystem(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, disk string) error {
+	// The PVC info is a nice to have
+	pvcName := os.Getenv(osd.PVCNameEnvVarName)
+	if pvcName == "" {
+		return errors.Errorf("failed to find PVC_NAME environment variable")
+	}
+	subsystem := fmt.Sprintf("ceph_fsid=%s", clusterInfo.FSID)
+	label := fmt.Sprintf("pvc_name=%s", pvcName)
+
+	logger.Infof("setting luks subsystem to %q and label to %q to disk %q", subsystem, label, disk)
+	args := []string{"config", disk, "--subsystem", subsystem, "--label", label}
+	output, err := context.Executor.ExecuteCommandWithCombinedOutput(cryptsetupBinary, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set subsystem %q and label %q to encrypted device %q. %s", subsystem, label, disk, output)
+	}
+
+	logger.Infof("successfully set luks subsystem to %q and label to %q to disk %q", subsystem, label, disk)
+	return nil
+}
+
+func dumpLUKS(context *clusterd.Context, disk string) (string, error) {
+	args := []string{"luksDump", disk}
+	cryptsetupOut, err := context.Executor.ExecuteCommandWithCombinedOutput(cryptsetupBinary, args...)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to close encrypted device. %s", cryptsetupOut)
+	}
+
+	return cryptsetupOut, nil
+}
+
+func isCephEncryptedBlock(context *clusterd.Context, currentClusterFSID string, disk string) bool {
+	metadata, err := dumpLUKS(context, disk)
+	if err != nil {
+		logger.Errorf("failed to determine if the encrypted block %q is from our cluster. %v", disk, err)
+		return false
+	}
+
+	// Now we parse the CLI output
+	// JSON output is only available with cryptsetup 2.4.x - https://gitlab.com/cryptsetup/cryptsetup/-/issues/511
+	ceph_fsid := luksLabelCephFSID.FindString(metadata)
+	if ceph_fsid == "" {
+		logger.Error("failed to find ceph_fsid in the LUKS header, the encrypted disk is not from a ceph cluster")
+		return false
+	}
+
+	// is it an OSD from our cluster?
+	currentDiskCephFSID := strings.SplitAfter(ceph_fsid, "=")[1]
```

Comment:
Any chance an "=" won't be found in the string?

Reply:
Very unlikely, this would mean `clusterInfo.FSID` is not initialized.